proposed/2022/dotnet-nuget-config-spec.md

```diff
@@ -0,0 +1,358 @@
+
+```
+dotnet nuget config get
+
+<configuration>
+  <packageSources>
+    <add key="source1" value="https://test/source1/v3/index.json" />
+    <add key="source2" value="https://test/source2/v3/index.json" />
+  </packageSources>
+  <packageSourceMapping>
+    <clear />
+    <packageSource key = "source1">
+      <package pattern="microsoft.*" />
+      <package pattern="nuget.*" />
+    </packageSource>
+    <packageSource key = "source2">
+      <package pattern="system.*" />
+    </packageSource>
+  </packageSourceMapping>
+  <packageRestore>
+    <add key="enabled" value="False" />
+    <add key="automatic" value="False" />
+  </packageRestore>
+</configuration>
+
+```
+- Get all the NuGet configuration settings that will be applied, when invoking NuGet command in the specific directory.
+
+```
+dotnet nuget config get --working-dir C:\Test\Repos
+
+<configuration>
+  <packageSources>
+    <add key="source1" value="https://test/source1/v3/index.json" />
+    <add key="source2" value="https://test/source2/v3/index.json" />
+  </packageSources>
+  <packageSourceMapping>
+    <clear />
+    <packageSource key = "source1">
+      <package pattern="microsoft.*" />
+      <package pattern="nuget.*" />
+    </packageSource>
+    <packageSource key = "source2">
+      <package pattern="system.*" />
+    </packageSource>
+  </packageSourceMapping>
+  <packageRestore>
+    <add key="enabled" value="False" />
+    <add key="automatic" value="False" />
+  </packageRestore>
+</configuration>
+
+```
+
+- Get all the NuGet configuration settings that will be applied, when invoking NuGet command in the current directory. Show the source(nuget configuration file path) of each configuration settings/child items.
+
+```
+dotnet nuget config get -v d
+
+<configuration>
+  <packageSources>
+    <add key="source1" value="https://test/source1/v3/index.json" />   <!-- file: C:\Test\Repos\Solution\NuGet.Config -->
+    <add key="source2" value="https://test/source2/v3/index.json" />   <!-- file: C:\Test\Repos\NuGet.Config -->
+  </packageSources>
+  <packageSourceMapping>
+    <clear />                                 <!-- file: C:\Users\username\AppData\Roaming\NuGet\NuGet.Config -->
+    <packageSource key = "source1">           <!-- file: C:\Test\Repos\Solution\NuGet.Config -->
+      <package pattern="microsoft.*" />
+      <package pattern="nuget.*" />
+    </packageSource>
+    <packageSource key = "source2">           <!-- file: C:\Test\Repos\NuGet.Config -->
+      <package pattern="system.*" />
+    </packageSource>
+  </packageSourceMapping>
+</configuration>
+
+```
+
+- Get `http_proxy` from config section, when invoking NuGet command in the current directory.
+
+```
+dotnet nuget config get http_proxy 
+
+http://company-squid:3128@contoso.com"
+
+```
+
+- Get `http_proxy` from config section, when `http_proxy` is not set in any of the NuGet configuration files.
+
+```
+dotnet nuget config get http_proxy 
+
+Key 'http_proxy' not found.
+
+```
```

Comment:
I think we're missing a more natural/more valuable idea. 

`dotnet nuget config get http_proxy --verbosity detailed`  could be: 
http://company-squid:3128@contoso.com" - C:\Users\username\AppData\Roaming\NuGet\NuGet.Config

I think this is is way more valuable than the  `dotnet nuget config get` xml format.

Reply:
Added this as an example. Thanks!